### ISSUE TITLE
Navigation: Dropdown Entry and Exit Animations

### ISSUE THREAD

**Issue by tochwill** [bbc/simorgh#4579]
**Is your feature request related to a problem? Please describe.**
The designs for the navigation dropdown include animations on show and hide:

<img width="565" alt="Screenshot 2019-11-06 at 14 39 03" src="https://user-images.githubusercontent.com/6690955/68311782-f77f2780-00a9-11ea-99fd-154d7d357bec.png">

<img width="571" alt="Screenshot 2019-11-06 at 14 39 12" src="https://user-images.githubusercontent.com/6690955/68311789-fa7a1800-00a9-11ea-8b42-c61fda08813b.png">

**Describe the solution you'd like**
- Canonical - animate the dropdown in on mount, and out on unmount. The official react package for this is: https://reactjs.org/docs/animation.html#high-level-api-reactcsstransitiongroup. If considering alternatives, ensure they are not too heavy.

When considering which properties to animate, ensure that on a lower powered device (@tochwill check what we class this as - currently testing on chrome with 6x processing slowdown enabled) the animation works and isn't janky. Dev and test can pair on testing this. Also consider Opera Mini.

We can create a component to handle the animation, and wrap the components to be animated within this when composing the navigation.

- AMP - We can consider an approach such as the following: https://amp.dev/documentation/guides-and-tutorials/develop/animations/triggering_css_animations/?format=websites.

When picking up this story, pair with another dev and discuss animation approaches upfront.

There is a video of the animation design in #ws-frontpages_ux-dev.

**Describe alternatives you've considered**

**Testing notes**
Ensure when animations are turned off by the user using a system wide setting (e.g. reduce motion in system preferences), that these are adhered to.

Check animation speed works for shortest and longest navs - can decide is we need to parameterise this after this test.

Dev insight: Will Cypress tests be required or are unit tests sufficient? Will there be any potential regression? etc

Ensure cypress tests take into account that the navigation will be animating.

- [x] This feature is expected to need manual testing.

**Additional context**

**Comment by tochwill** [comment]
Blocked on https://github.com/bbc/simorgh/issues/4571 and https://github.com/bbc/simorgh/issues/4570

**Comment by AlistairGempf** [comment]
Some initial amp animation findings:
- We can use either CSS animations or `amp-animation` components
- `amp-animation` is can only animate on [these properties](https://amp.dev/documentation/components/amp-animation/#white-listed-properties-for-keyframes):
  - opacity
  - transform
  - visibility
  - offset-distance
- We may be able to use the `transform` `amp-animation`, but would need to apply it both to the dropdown menu element and the whole body
- With CSS animation we cannot animate to or from an `auto` height
- So in either case, we need to know the height of the dropdown menu
  - This is not too tricky with javascript running in the window, but in amp is going to be more difficult
  - Should be possible to do during the SSR based on the `line-height`, the `padding` and the number of elements in the nav
    - [This will be ugly though](https://github.com/bbc/simorgh/compare/integrate-new-navigation...test-amp-navigation#diff-c9bc01231bfbbecdd7ac932273c0df80R33)
    - Would need to be calculated in `rem`
- We should also investigate how performant `height` animation and `transform` animations are
- We also need to find a way to delay the `toggleVisibility` taking effect
  - One possiblity is using a class to hide and show the element using css and giving us a bit more control

**Comment by AlistairGempf** [comment]
Some `amp-animation` implementation details:
- The docs say you can trigger the animation based on `visibility` however that really means when the element scrolls into view
  - We will therefore need to add a `animation-id.start` to the `on` attribute of the button
  - However this would trigger the same animation for opening and closing
    - One solution could be having an `onClose` and an `onOpen` prop and binding the `on` attribute to the menu-expanded state
- The docs talk about having access to the `height()` of the element
  - This is useful except the `height()` of the element when it is hidden is 0
     - When it is unhidden then the `height()` gives the height of the dropdown

[Here](https://github.com/bbc/simorgh/compare/integrate-new-navigation...test-amp-navigation) is an example `amp-animation` implementation with the boilderplate code that will need to be added.

**Comment by tochwill** [comment]
Thanks so much for the POC into AMP animation @AlistairGempf - as we have quite a lot of things we still need to figure out, we're ok to descope animations on the AMP site for the initial release. I'll add another ticket to the board that we can pick up afterwards to continue the investigation.

**Comment by tochwill** [comment]
Have opened #4989 to deal with AMP animation

**Comment by PriyaKR** [comment]
Can someone add the design docs link pls.

**Comment by PriyaKR** [comment]
https://app.zeplin.io/project/5c8a5e021d454124ef90669f/dashboard?tag=Navigation

**Comment by OlgaLyubin** [comment]
Canonical animation has been done in https://github.com/bbc/psammead/pull/2847. There is an issue opened for AMP animation https://github.com/bbc/simorgh/issues/4989 which will be picked up at later stages since the new navigation will go live without an AMP animation at first.
